src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -57,6 +58,13 @@
     
     abstract public SamReader open(final File file);
 
+    public SamReader open(final Path path) {
+        final SamInputResource r = SamInputResource.of(path);
+        final Path indexMaybe = SamFiles.findIndex(path);
+        if (indexMaybe != null) r.index(indexMaybe);
+        return open(r);
+    }
```

Comment:
What I mean, though, is that `referenceSequence()` takes a `File` -- there is no overload that takes a `Path`. If you're not going to address this here, can you open a ticket for it?